lib/action_controller/serialization.rb

```diff
@@ -6,7 +6,7 @@ module Serialization
 
     include ActionController::Renderers
 
-    ADAPTER_OPTION_KEYS = [:include, :fields, :root, :adapter]
+    ADAPTER_OPTION_KEYS = [:include, :fields, :adapter]
```

Comment:
Yeah @bf4 this is indeed a big change.
The idea is that ```root``` shouldn't be an option, but another adapter by itself. 
This make it even easier to integrate with some js frameworks.
We also removed the option to set a specific ```root``` key, so the root isn't "going" to the serializer, but is the serializer the one that knows what it should be and the adapter the one the uses it or not.